### ISSUE TITLE
Avoid 1:...; false positive on hard-coded ranges

### ISSUE THREAD

**Issue by plantarum** [Bioconductor/BiocCheck#29]
I understand the rationale behind discouraging `1:variable` or `1:function(variable)`, but there are some instances where I know that I want the first (or last, or ...) n elements of a vector or list. In my code I have:

```
  ## remove the first 5 bins, eliminating noisy artifacts produced by
  ## instrument compensation. This is below the level of actual data, so
  ## shouldn't cause any problems with analysis.
  intensity[1:5] <- 0
```

This raises the flag, suggesting you'd prefer I use `intensity[seq_len(5)]`? I think in this case `1:5` is both safe and more easily understood than the alternative? Is it possible to exclude such cases from this rule?

**Comment by lshep** [comment]
In general it is still better practice when subset vectors or lists to have
named vectors and lists and subset accordingly rather than relying on arbitrary
ordering. We won't change the currently implementation as in the majority of
cases it is better practice to use seq_len to catch unforeseen edge cases.  It
will be up to the reviewer of your package to decide if your use case would
warrant an exception. Please give the above argument in the discussion on your
open issue in the tracker.
